docs/modeling/example-fitting-model-sets.rst

```diff
@@ -1,37 +1,120 @@
-      >>> print(new_model)  # doctest: +SKIP
-      Model: Polynomial1D
-      Inputs: 1
-      Outputs: 1
-      Model set size: 2
-      Degree: 3
-      Parameters:
-           c0  c1         c2                 c3
-          --- --- ------------------ -----------------
-          1.0 2.0 -5.86673908219e-16 3.61636197841e-17
-          1.0 2.0 -5.86673908219e-16 3.61636197841e-17
+In the example below, we create a 3D data cube where the first dimension is a ramp, 
+for example as from non-destructive readouts of an IR detector. So each pixel has a 
+depth along a time axis, and flux that results a total number of counts that is 
+increasing with time. We will be fitting a 1D polynomial vs. time to estimate the 
+flux in counts/second (the slope of the fit).
+
+   .. plot::
+       :include-source:
+
+       import numpy as np
+       import matplotlib.pyplot as plt
+       from scipy import stats
+       from astropy.modeling import models, fitting
+       
+       # Set up the shape of the image and create the time axis
+       DEPTH,WIDTH,HEIGHT=10,101,102 # Time is along the DEPTH axis
+       t = np.arange(DEPTH,dtype=np.float64)*10. # e.g. readouts every 10 seconds
+       
+       # Make up a flux in each pixel 
+       fluxes = np.arange(1.*WIDTH*HEIGHT).reshape(HEIGHT,WIDTH)
+       # Create the ramps by integrating the fluxes along the time steps
+       image = np.stack([fluxes*dt for dt in t],axis=0)
+       # Add some Gaussian noise to each sample
+       image += stats.norm.rvs(0.,image*0.05,size=image.shape) # Add noise
+       print("The image shape is: ",image.shape)
+       
+       # Create the models and the fitter
+       N=WIDTH*HEIGHT # This is how many instances we need
+       line=models.Polynomial1D(degree=1,n_models=N)
+       fit = fitting.LinearLSQFitter()
+       print("We created %d models" % len(line))
+       
+       # We need to get the data to be fit into the right shape
+       # In this case, the time axis can be one dimensional.
+       # The fluxes have to be organized into an array 
+       # that is of shape `(WIDTH*HEIGHT,DEPTH)`
+       # i.e we are reshaping to flatten last two axes and 
+       # transposing to put them first.
+       pixels = image.reshape((DEPTH,WIDTH*HEIGHT))
+       y = pixels.T
+       print("x axis is one dimensional: ",t.shape)
+       print("y axis is two dimensional, N by len(x): ", y.shape)
+       
+       # Fit the model. It does the looping over the N models implicitly
+       new_model = fit(line,x=t,y=y)
+       print("We fit %d models: " % len(new_model))
+       
+       # Fill an array with values computed from the best fit and reshape it to match the original
+       best_fit = new_model(t,model_set_axis=False).T.reshape((DEPTH,HEIGHT,WIDTH))
+       print("We reshaped the best fit to dimensions: ",best_fit.shape)
+       
+       # Plot the fit along a couple of pixels
+       plt.figure(figsize=(10,5))
+       plt.subplot(121)
+       X,Y=HEIGHT//5,WIDTH//5
```

Comment:
What does 5 mean?

Reply:
Yeah, that's probably too obscure. I was just trying to make a plot for a couple random pixels within the array. I'll do it differently.

Reply:
Just hardcoded in the pixel values; this is an example after all.